### PR TITLE
Dockerfile

### PR DESCRIPTION
**Pull request by DS-KoolAid**
Adding Dockerfile and the associated documentation to the README.md

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM rust:1.65 as build
+RUN apt update -y
+RUN apt install build-essential libhyperscan-dev pkg-config git -y
```
Comment (bradlarsen) [inline Dockerfile:3]: You'll want to combine these two lines into a single `RUN` command for this to behave as expected with Docker caching. See https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run

```suggestion
RUN apt update && apt install -y \
    build-essential \
    libhyperscan-dev \
    pkg-config git
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+WORKDIR "/noseyparker"
+COPY . .
+RUN cargo build --release
+
+FROM debian:11-slim
+COPY --from=build /noseyparker/target/release/noseyparker /usr/bin/noseyparker
+RUN apt update -y
+RUN apt install libhyperscan-dev -y
```
Comment (bradlarsen) [inline Dockerfile:11]: You'll want to combine these two lines into a single `RUN` command for this to behave as expected with Docker caching. See https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run

```suggestion
RUN apt update && apt install -y \
    libhyperscan-dev
```

**Review by bradlarsen — CHANGES_REQUESTED** [review]
Thanks @DS-KoolAid! Great contribution. I like the multi-stage build approach, which should leave us with a pretty small final image to actually run.

I will merge this after a couple Docker nits are addressed. I suggested a couple fixes that I think you could just click to apply -- I hope I spelled them right.

**Comment by DS-KoolAid** [comment]
Changes committed!

**Review by bradlarsen — APPROVED** [review]

**Pull request merged**